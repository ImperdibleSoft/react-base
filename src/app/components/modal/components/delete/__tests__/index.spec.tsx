import * as React from 'react';
import { shallow } from 'enzyme';

import DeleteModal from '..';

describe('DeleteModal', () => {
  const modalRoot = document.createElement('div');
  modalRoot.setAttribute('id', 'modal-root');
  document.body.appendChild(modalRoot);

  const handleClose = () => {};
  const handleSubmit = () => {};

  it('should render a closed modal', () => {
    const component = shallow(
      <DeleteModal handleClose={handleClose} handleSubmit={handleSubmit} isFetching={false} isOpen={false}>
        <p>Test children</p>
      </DeleteModal>
    );

    expect(component).toMatchSnapshot();
  });

  it('should render an empty open modal', () => {
    const component = shallow(
      <DeleteModal handleClose={handleClose} handleSubmit={handleSubmit} isFetching={false} isOpen={true}></DeleteModal>
    );

    expect(component).toMatchSnapshot();
  });

  it('should render an open fetching modal', () => {
    const component = shallow(
      <DeleteModal handleClose={handleClose} handleSubmit={handleSubmit} isFetching={true} isOpen={true}>
        <p>Test children</p>
      </DeleteModal>
    );

    expect(component).toMatchSnapshot();
  });

  it('should render an open modal', () => {
    const component = shallow(
      <DeleteModal handleClose={handleClose} handleSubmit={handleSubmit} isFetching={false} isOpen={true}>
        <p>Test children</p>
      </DeleteModal>
    );

    expect(component).toMatchSnapshot();
  });
});
