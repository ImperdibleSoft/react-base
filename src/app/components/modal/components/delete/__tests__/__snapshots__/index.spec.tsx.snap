// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteModal should render a closed modal 1`] = `
<Modal
  handleClose={[Function]}
  isOpen={false}
>
  <Header>
    <h5>
      Delete
    </h5>
  </Header>
  <Body>
    <p>
      Test children
    </p>
  </Body>
  <Body>
    <p>
      This action 
      <b>
        cannot be undone
      </b>
      .
    </p>
  </Body>
  <Footer>
    <Buttons
      options={
        Array [
          Object {
            "buttonType": "danger",
            "icon": "trash",
            "id": "confirm-delete",
            "isDisabled": false,
            "label": "Delete",
            "onClick": [Function],
            "type": "button",
          },
          Object {
            "icon": "times",
            "id": "cancel-delete",
            "isDisabled": false,
            "label": "Cancel",
            "onClick": [Function],
            "type": "button",
          },
        ]
      }
    />
  </Footer>
</Modal>
`;

exports[`DeleteModal should render an empty open modal 1`] = `
<Modal
  handleClose={[Function]}
  isOpen={true}
>
  <Header>
    <h5>
      Delete
    </h5>
  </Header>
  <Body>
    <p>
      Are you sure you want to delete this?
    </p>
  </Body>
  <Body>
    <p>
      This action 
      <b>
        cannot be undone
      </b>
      .
    </p>
  </Body>
  <Footer>
    <Buttons
      options={
        Array [
          Object {
            "buttonType": "danger",
            "icon": "trash",
            "id": "confirm-delete",
            "isDisabled": false,
            "label": "Delete",
            "onClick": [Function],
            "type": "button",
          },
          Object {
            "icon": "times",
            "id": "cancel-delete",
            "isDisabled": false,
            "label": "Cancel",
            "onClick": [Function],
            "type": "button",
          },
        ]
      }
    />
  </Footer>
</Modal>
`;

exports[`DeleteModal should render an open fetching modal 1`] = `
<Modal
  handleClose={[Function]}
  isOpen={true}
>
  <Header>
    <h5>
      Delete
    </h5>
  </Header>
  <Body>
    <p>
      Test children
    </p>
  </Body>
  <Body>
    <p>
      This action 
      <b>
        cannot be undone
      </b>
      .
    </p>
  </Body>
  <Footer>
    <Buttons
      options={
        Array [
          Object {
            "buttonType": "danger",
            "icon": "trash",
            "id": "confirm-delete",
            "isDisabled": true,
            "label": "Delete",
            "onClick": [Function],
            "type": "button",
          },
          Object {
            "icon": "times",
            "id": "cancel-delete",
            "isDisabled": true,
            "label": "Cancel",
            "onClick": [Function],
            "type": "button",
          },
        ]
      }
    />
  </Footer>
</Modal>
`;

exports[`DeleteModal should render an open modal 1`] = `
<Modal
  handleClose={[Function]}
  isOpen={true}
>
  <Header>
    <h5>
      Delete
    </h5>
  </Header>
  <Body>
    <p>
      Test children
    </p>
  </Body>
  <Body>
    <p>
      This action 
      <b>
        cannot be undone
      </b>
      .
    </p>
  </Body>
  <Footer>
    <Buttons
      options={
        Array [
          Object {
            "buttonType": "danger",
            "icon": "trash",
            "id": "confirm-delete",
            "isDisabled": false,
            "label": "Delete",
            "onClick": [Function],
            "type": "button",
          },
          Object {
            "icon": "times",
            "id": "cancel-delete",
            "isDisabled": false,
            "label": "Cancel",
            "onClick": [Function],
            "type": "button",
          },
        ]
      }
    />
  </Footer>
</Modal>
`;
